nequip/train/trainer.py

```diff
@@ -1144,12 +1144,55 @@ def __del__(self):
         for i in range(len(logger.handlers)):
             logger.handlers.pop()
 
+    def _parse_n_train_n_val(
+        self, train_dataset_size: int, val_dataset_size: int
+    ) -> tuple[int]:
+        # parse n_train and n_val (can be ints or str with percentage):
+        n_train_n_val = []
+        for n_name, dataset_size in (
+            ("n_train", train_dataset_size),
+            ("n_val", val_dataset_size),
+        ):
+            n = getattr(self, n_name)
+            if isinstance(n, str) and "%" in n:
+                n_train_n_val.append(
+                    (float(n.rstrip("%")) / 100) * dataset_size
+                )  # convert to float first
+            elif isinstance(n, int):
+                n_train_n_val.append(n)
+            else:
+                raise ValueError(
+                    f"Invalid value/type for {n_name}: {n} -- must be either int or str with %!"
+                )
+
+        floored_n_train_n_val = [int(n) for n in n_train_n_val]
```

Comment:
```suggestion
        floored_n_train_n_val = [int(n) for n in n_train_n_val]
        for n in floored_n_train_n_val:
            assert n > 0
```

Reply:
This check has been added in [4de79c3](https://github.com/mir-group/nequip/pull/438/commits/4de79c3faf0d0c8f9a88efeebe35fb397e28b7d8) along with an error message if it fails